### PR TITLE
SceneBuilder should build a tree of layers

### PR DESCRIPTION
**Pull request by abarth**
We now build a layer tree before squashing the draw command down to an
SkPicture.

### CODE REVIEW

**Comment by chinmaygarde** [comment]
lgtm

File: sky/compositor/layer.h [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+  ContainerLayer();
+  ~ContainerLayer() override;
+
+  void Add(PassOwnPtr<Layer> layer);
+  const Vector<OwnPtr<Layer>>& layers() const { return layers_; }
+
+  void PaintChildren(SkCanvas* canvas);
+
+ private:
+  Vector<OwnPtr<Layer>> layers_;
```
Comment (jamesr) [inline sky/compositor/layer.h]: why Vector<OwnPtr<>> instead of std::vector<std::unique_ptr<>> ?

File: sky/compositor/layer.h [hunk 2]
```diff
@@ -0,0 +1,155 @@
...
+
+  virtual void Paint(SkCanvas* canvas) = 0;
+
+  ContainerLayer* parent() const { return parent_; }
+  void set_parent(ContainerLayer* parent) { parent_ = parent; }
+
+  const SkRect& paint_bounds() const { return paint_bounds_; }
+  void set_paint_bounds(const SkRect& paint_bounds) { paint_bounds_ = paint_bounds; }
+
+ private:
```
Comment (jamesr) [inline sky/compositor/layer.h]: you should probably explicitly disallow copy/assignment

**Comment by abarth** [comment]
Will do.

**Pull request merged**